gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/ManagedController.java

```diff
@@ -247,19 +247,17 @@ void createPin(
     keelService.pin(application, pin);
   }
 
-  @ApiOperation(value = "Delete a pin for an artifact in an environment")
-  @DeleteMapping(path = "/application/{application}/pin")
-  void deletePin(
-      @PathVariable("application") String application, @RequestBody EnvironmentArtifactPin pin) {
-    keelService.deletePin(application, pin);
-  }
-
   @ApiOperation(
-      value = "Delete a pin for an artifact in an environment by specifying the environment")
+      value =
+          "Unpin one or more artifact in an environment. If the `reference` parameter is specified, only "
+              + "the corresponding artifact will be unpinned. If it's omitted, all pinned artifacts in the environment will be "
+              + "unpinned.")
   @DeleteMapping(path = "/application/{application}/pin/{targetEnvironment}")
-  void deletePinForEnv(
-      @PathVariable("application") String application, @PathVariable String targetEnvironment) {
-    keelService.deletePinForEnvironment(application, targetEnvironment);
+  void deletePin(
+      @PathVariable("application") String application,
+      @PathVariable("targetEnvironment") String targetEnvironment,
+      @RequestParam(value = "reference", required = false) String reference) {
```

Comment:
do we know of any non-reference usage of this endpoint as of today? I think in an ideal world I'd advocate for making this mandatory, at least at the gate level, but if not I don't think it's a big deal.

Reply:
I thought the UI you were building didn't use the reference because it was broken? 🙂 
Let's get this merged and I'll cleanup later based on our conclusions!